sktime/forecasting/model_selection/_split.py

```diff
@@ -388,39 +457,59 @@ class CutoffSplitter(BaseSplitter):
     Parameters
     ----------
     cutoffs : np.array or pd.Index
-        cutoff points, positive and integer-index like, usable with pandas
```

Comment:
I think it is crucial to describe how the splits are defined, for a given cutoff. Be precise in the docstring.

For instance, is the test set comprised of all data with bigger-equal, or strictly-bigger-equal time index?

Reply:
another detail is if you pass a `pandas.Series` with integer indices, but the indices are not `RangeIndex(len(y))`. On can imagine two behaviours here, `loc` and `iloc` based. Which one applies? What if the user wants the other one? Would adding a flag with a default make sense (to constructor, perhaps)?

Reply:
added the following sentence:
```
Training window's last point is equal to the cutoff,
while test window starts from the next observation in `y`.
```

Reply:
> another detail is if you pass a `pandas.Series` with integer indices, but the indices are not `RangeIndex(len(y))`. On can imagine two behaviours here, `loc` and `iloc` based. Which one applies? What if the user wants the other one? Would adding a flag with a default make sense (to constructor, perhaps)?

Now I really find this sentence confusing. I have removed "usable with pandas `.iloc[]` or `.loc[]` indexing, respectively" everywhere since neither `loc` nor `iloc` are actually used in splitters to construct windows. A user implicitly specifies what indexing he/she wants by choosing the type of e.g. `cutoffs`. It is our task to figure out how to treat user's choice. Some choices are clearly forbidden (`TypeError`).